src/main/java/com/google/codeu/servlets/ChartsDataServlet.java

```diff
@@ -1,132 +1,196 @@
 package com.google.codeu.servlets;
 
 import java.io.IOException;
-import java.lang.reflect.Type;
+import java.time.Duration;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.Arrays;
+import java.util.Date;
 import java.util.List;
+import java.util.UUID;
 
 import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
-import com.google.gson.JsonArray;
-import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
 
 @WebServlet("/chartsdata/*")
 public class ChartsDataServlet extends HttpServlet{
 	
-	private JsonArray userMealArray;
+	private DatastoreService datastore;
 
 	/*
 	 * Class that models the JSON for our dummy data.
 	 * carbonFootprint will eventually be calculated from the map of FoodItems
 	 * For now, it is assigned a random value
 	 * */
-	private static class UserMeal {
-		double carbonFootprint;
-		LocalDate date;
+	private static class EatenMeal {
+		private UUID id;
+		private List<String> foods;
+		private List<Double> amounts;
+		private Date date;
+		private double footprint;
 
-		private UserMeal(double carbonFootprint, LocalDate date) {
-			this.carbonFootprint = carbonFootprint;
+		private EatenMeal(UUID id, List<String> foods, List<Double> amounts, Date date) {
```

Comment:
has this been rebased since my pr was merged? I think i changed the foods and amounts from lists to just their base class types (string and double) as we only take one food in per form